modules/workflow-workflowoperation/src/test/java/org/opencastproject/workflow/handler/workflow/DuplicateEventWorkflowOperationHandlerTest.java

```diff
@@ -0,0 +1,296 @@
+
+    // run the operation handler
+    WorkflowOperationResult result = getWorkflowOperationResult(mp, configurations);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Track track = clonedMediaPackages.getValue().getTracksByTag("tag1")[0];
+    Assert.assertEquals("tag1", track.getTags()[0]);
+    Assert.assertEquals("tag2", track.getTags()[1]);
+  }
+
+  @Test
+  public void testRemoveAndAddTags() throws Exception {
+    mockDependencies(1);
+    Map<String, String> configurations = new HashMap<>();
+    configurations.put(SOURCE_FLAVORS_PROPERTY, "*/*");
+    configurations.put(SOURCE_TAGS_PROPERTY, "part1");
+    configurations.put(TARGET_TAGS_PROPERTY, "-part1,+tag3");
+    configurations.put(NUMBER_PROPERTY, "" + 1);
+    configurations.put(MAX_NUMBER_PROPERTY, "" + 10);
+    configurations.put(PROPERTY_NAMESPACES_PROPERTY, "org.opencastproject.assetmanager.security");
+    configurations.put(COPY_NUMBER_PREFIX_PROPERTY, "copy");
+
+    // run the operation handler
+    WorkflowOperationResult result = getWorkflowOperationResult(mp, configurations);
+    Assert.assertEquals(Action.CONTINUE, result.getAction());
+
+    Track track = clonedMediaPackages.getValue().getTracksByTag("tag3")[0];
+    final List<String> newTags = Arrays.asList(track.getTags());
+    final List<String> originalTags = Arrays.asList(mp.getTracksByTag("part1")[0].getTags());
+    Assert.assertEquals(originalTags.size(), newTags.size());
+    Assert.assertTrue(newTags.contains("tag3"));
+    Assert.assertFalse(newTags.contains("part1"));
+  }
+
+  private WorkflowOperationResult getWorkflowOperationResult(MediaPackage mp, Map<String, String> configurations)
+      throws WorkflowOperationException {
+    // Add the mediapackage to a workflow instance
+    WorkflowInstanceImpl workflowInstance = new WorkflowInstanceImpl();
+    workflowInstance.setId(1);
+    workflowInstance.setState(WorkflowState.RUNNING);
+    workflowInstance.setMediaPackage(mp);
+    WorkflowOperationInstanceImpl operation = new WorkflowOperationInstanceImpl("op", OperationState.RUNNING);
+    operation.setTemplate("create-event");
+    operation.setState(OperationState.RUNNING);
+    for (String key : configurations.keySet()) {
+      operation.setConfiguration(key, configurations.get(key));
+    }
+
+    List<WorkflowOperationInstance> operationsList = new ArrayList<>();
+    operationsList.add(operation);
+    workflowInstance.setOperations(operationsList);
+
+    // Run the media package through the operation handler
+    return operationHandler.start(workflowInstance, null);
+  }
+
+  private void mockDependencies(int numberOfCopies) throws Exception {
+    clonedMediaPackages = Capture.newInstance(CaptureType.ALL);
+    reset(workspace, assetManager, distributionService);
+
+    URI uriDc = getClass().getResource("/dublincore.xml").toURI();
+    expect(workspace.read(eq(URI.create("dublincore.xml")))).andReturn(new File(uriDc)).times(numberOfCopies);
```

Comment:
This line conflicts with an earlier merged patch since `Workspace::read(…)` now returns an `InputStream` which causes a build to fail after it is merged into `develop`.

Reply:
Thanks, this should be fixed now